src/OTCNexus.sol

```diff
@@ -0,0 +1,283 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./OtcMath.sol";
+
+error Router__InvalidTokenAmount();
+error Router__InvalidPriceAmount();
+error Router__InvalidPriceMultiplier();
+error Router__InvalidDeadline();
+error Router__InvalidTokenAddresses();
+error Router__InvalidTokenAndAmountMismatch();
+error Router__AllowanceToken0TooLow();
+error Router__AllowanceToken1TooLow();
+error Router__ApproveToken0Failed();
+error Router__DepositToken0Failed();
+error Router__RfsRemoved();
+error Router__Amount1TooHigh();
+error Router__TransferToken0Failed();
+error Router__TransferToken1Failed();
+error Router__NotMaker();
+error Router__RefundToken0Failed();
+error Router__OnlyFixedOrDynamicAllowed();
+error Router__OnlyFixedPriceOrAmountAllowed();
+error Router__OnlyOneTokenAccepted();
+
+/**
+ * @title OTCNexus
+ * @notice core swap functionality
+ */
+contract OTCNexus is Ownable {
+    uint256 public rfsIdCounter;
+    mapping(uint256 => RFS) private idToRfs;
+    mapping(address => RFS[]) private makerRfs;
+
+    struct RFS {
+        uint256 id;
+        address maker;
+        address token0;
+        address[] tokensAccepted;
+        uint256 amount0;
+        uint256 amount1;
+        uint256 usdPrice; // Implies the fixed price per token0
+        uint256 priceMultiplier;
+        uint256 deadline;
+        bool isDynamic;
+        bool isDeposited;
+        bool removed;
+    }
+
+    event RfsUpdated(uint256 rfsId, uint256 amount0, uint256 amount1, uint256 deadline);
+    event RfsFilled(uint256 rfsId, address taker, uint256 amount0, uint256 amount1);
+    event RfsRemoved(uint256 rfsId);
+    event RfsCreated(
+        uint256 rfsId,
+        address maker,
+        address indexed token0,
+        address[] indexed tokensAccepted,
```

Comment:
The current solution lowers gas cost by a lot, since it just need to put into storage just 1 RFS. If that person would have to create various RFS, first you would have to check multiple time different amount0, plus n RFS would be put in storage. n times creation costs + n times deletion costs

Reply:
Ok, let it stay